src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketInformation.cs

```diff
@@ -8,5 +8,16 @@ public struct SocketInformation
     {
         public byte[] ProtocolInformation { get; set; }
         public SocketInformationOptions Options { get; set; }
+
+        internal void SetOption(SocketInformationOptions option, bool value)
+        {
+            if (value) Options |= option;
+            else Options &= ~option;
+        }
+
+        internal bool GetOption(SocketInformationOptions option)
+        {
+            return (Options & option) != 0;
```

Comment:
I would use `== option` here. It will be safer if `option` uses multiple bits.

The JIT is smart here and will output an optimal `TEST` instruction (equiv to `!= 0`) if the constant is a single bit.